### PR TITLE
reinit! VBT for proper reproducibility

### PR DESCRIPTION
**Pull request by rmsrosa**
This just adds the broken test related to issue #123 

I tagged it as draft because a fix is under way.

### CODE REVIEW

**Comment by rmsrosa** [comment]
Ok, this should close #123

**Comment by rmsrosa** [comment]
I started to work on backward integration with VBT, but I will need more time for that. Let's just fix the `reinit` for now, and I will work on a separate PR for the backward integration.

**Comment by rmsrosa** [comment]
Ok, I fixed the interpolation check in VBT so integration backwards seems to be working fine. This also closes #125. I also added a reproducibility test for `reinit` with backward integration. Local tests are fine. If CI passes, this is good to go.

**Comment by rmsrosa** [comment]
Just adding a final forward vs backward reproducibility test for VBT.

File: src/noise_interfaces/virtual_brownian_tree_interface.jl [hunk 1]
```diff
@@ -111,14 +116,15 @@ function accept_step!(W::VirtualBrownianTree, dt, u, p, setup_next = true)
...
     if (W.dt isa Union{Rational, Integer})
-        if sign(W.dt) * (W.curt + W.dt) > sign(W.dt) * W.t[end]
+        if t < t_min || t > t_max
             setup_next = false
             W.step_setup = false
         end
     else
-        if sign(W.dt) * (W.curt + W.dt) >
-           sign(W.dt) * (W.t[end] + sign(W.dt) * 10eps(typeof(dt)))
+        if t < t_min - 10eps(typeof(dt)) || t > t_max + 10eps(typeof(dt))
```
Comment (ChrisRackauckas) [inline src/noise_interfaces/virtual_brownian_tree_interface.jl:127]: much nicer 😅

**Comment by rmsrosa** [comment]
This closes #123 and #125 as I wrote above.

Why it didn't close them automatically?

**Comment by ChrisRackauckas** [comment]
You have to put that in the first post, like "Closes #123" or "Fixes #123"

**Pull request merged**